### PR TITLE
Use span APIs to compute StringSegment hashes when available

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/Extensions/issues/457

### CODE REVIEW

File: src/Primitives/src/Microsoft.Extensions.Primitives.csproj [hunk 1]
```diff
@@ -1,11 +1,11 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <Description>Primitives shared by framework extensions. Commonly used types include:
 Microsoft.Extensions.Primitives.IChangeToken
 Microsoft.Extensions.Primitives.StringValues
 Microsoft.Extensions.Primitives.StringSegment</Description>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFrameworks>netstandard2.0;netcoreapp3.0</TargetFrameworks>
```
Comment (pranavkm) [inline src/Primitives/src/Microsoft.Extensions.Primitives.csproj]: cc @natemcmaster

Comment (natemcmaster) [reply src/Primitives/src/Microsoft.Extensions.Primitives.csproj]: Is this the final TFM, or will this be reverted to netstandard2.1 before we RTM?

cc @davidfowl

File: src/Primitives/src/StringSegmentComparer.cs [hunk 2]
```diff
@@ -34,8 +35,17 @@ public bool Equals(StringSegment x, StringSegment y)
...
         public int GetHashCode(StringSegment obj)
         {
-            // TODO: PERF; Note that .NET Core strings use randomized hash codes for security reasons.
+#if NETCOREAPP
+            return string.GetHashCode(obj.AsSpan(), Comparison);
+#else
+            if (!obj.HasValue)
+            {
+                return 0;
+            }
```
Comment (rynowak) [inline src/Primitives/src/StringSegmentComparer.cs]: Is this just being done for perf for the degenerate case?

Comment (pranavkm) [reply src/Primitives/src/StringSegmentComparer.cs]: The current code throws if you have a `StringSegment` instance without a backing buffer (e.g. `var x = default(StringSegment);`). I used existing test data for `StringSegment` which cover this case.

**Pull request merged**